sched/scheduler/jobRunner.go

```diff
@@ -82,6 +60,32 @@ func runJob(job sched.Job, saga *s.Saga, nodes []cm.Node) {
 	return
 }
 
+// Logic to execute a single task in a Job.  Ensures Messages are logged to SagaLog
+// Logs StartTask, executes Tasks, Logs EndTask
+func runTask(saga *s.Saga, node cm.Node, taskId string, task sched.TaskDefinition) {
+	// Put StartTask Message on SagaLog
+	stErr := saga.StartTask(taskId, nil)
+	if stErr != nil {
+		handleSagaLogErrors(stErr)
+	}
+
+	// TODO: After a number of attempts we should stop
+	// Trying to run a task, could be a poison pill
+	// Implement deadletter queue
+	taskExecuted := false
+	for !taskExecuted {
+		execErr := node.SendMessage(task)
+		if execErr == nil {
+			taskExecuted = true
```

Comment:
if you just break here you can skip having taskExecuted variable. If you want to keep it, I'd make it a 3-clause for loop (with an empty increment, for now)